File: backend/ruby_foundations/exercises/small_problems_101/easy_7/staggered_caps_1.rb
def staggered_case(string)
  result = string.chars.each_with_index do |letter, index|
    index.odd? ? letter.downcase! : letter.upcase!
  end
  result.join
end

puts staggered_case('I Love Launch School!') == 'I LoVe lAuNcH ScHoOl!'
puts staggered_case('ALL_CAPS') == 'AlL_CaPs'
puts staggered_case('ignore 77 the 444 numbers') == 'IgNoRe 77 ThE 444 NuMbErS'

File: frontend/javascript/exercises/string_and_text_processing/staggered_caps_1.js
function staggeredCase(string) {
  return string.split('').map(function (char, index) {
    if (index % 2 === 0) {
      return char.toUpperCase();
    } else {
      return char.toLowerCase();
    }
  }).join('');
}

console.log(staggeredCase('I Love Launch School!') === 'I LoVe lAuNcH ScHoOl!');     // 'I LoVe lAuNcH ScHoOl!'
console.log(staggeredCase('ALL_CAPS') === 'AlL_CaPs')                  // 'AlL_CaPs'
console.log(staggeredCase('ignore 77 the 444 numbers') === 'IgNoRe 77 ThE 444 NuMbErS'); // 'IgNoRe 77 ThE 444 NuMbErS'